gen_cert.py

```diff
@@ -1853,3 +1855,453 @@ def _generate_v3_dynamic_certificate(
+                         unset means to use the value passed in as part of the
+                         certificate request.
+        ISSUED_DATE    - (optional) If given, the date string which should be
+                         stamped onto each and every certificate. The value
+                         ROLLING is equivalent to leaving ISSUED_DATE unset, which
+                         stamps the certificates with the current date.
+        HAS_DISCLAIMER - (optional) If given, the programmatic disclaimer that
+                         is usually rendered at the bottom of the page, is not.
+        TEMPLATEFILE   - (optional) If given, the filename referred to by
+                         TEMPLATEFILE will be used as the template over which
+                         to render.
+
+        RETURNS (download_uuid, verify_uuid, download_url)
+        """
+        
+        verify_me_p = self.cert_data.get('VERIFY', True)
+        verify_uuid = uuid.uuid4().hex if verify_me_p else ''
+        download_uuid = uuid.uuid4().hex
+        download_url = "{base_url}/{cert}/{uuid}/{file}".format(
+            base_url=settings.CERT_DOWNLOAD_URL,
+            cert=S3_CERT_PATH,
+            uuid=download_uuid,
+            file=filename,
+        )
+
+        filename = os.path.join(download_dir, download_uuid, filename)
+
+        # This file is overlaid on the template certificate
+        overlay_pdf_buffer = StringIO.StringIO()
+        PAGE = canvas.Canvas(overlay_pdf_buffer, pagesize=landscape(A4))
+        
+        WIDTH, HEIGHT = landscape(A4)  # Width and Height of landscape canvas (in points)
+        
+        ##### Fonts #####
+        
+        # 0 0 - normal
+        # 0 1 - italic
+        # 1 0 - bold
+        # 1 1 - italic and bold
+        addMapping('OpenSans-Light', 0, 0, 'OpenSans-Light')
+        addMapping('OpenSans-Light', 1, 0, 'OpenSans-Bold')
+        addMapping('SourceSansPro-Regular', 0, 0, 'SourceSansPro-Regular')
+        addMapping('SourceSansPro-Regular', 1, 0, 'SourceSansPro-Bold')
+        addMapping('SourceSansPro-Regular', 1, 1, 'SourceSansPro-BoldItalic')
+
+        styleArial = ParagraphStyle(name="arial", fontName='Arial Unicode')
+        styleOpenSansLight = ParagraphStyle(name="opensans-light", fontName='OpenSans-Light')
+        styleSourceSansPro = ParagraphStyle(name="sourcesans-regular", fontName='SourceSansPro-Regular')
+        
+        # These are ordered by preference; cf. font_for_string() above
+        fontlist = [
+            ('SourceSansPro-Regular', 'SourceSansPro-Regular.ttf', None),
+            ('OpenSans-Light', 'OpenSans-Light.ttf', None),
+            ('Arial Unicode', 'Arial Unicode.ttf', None),
+        ]
+        
+        ##### Draw main rectangle containers #####
+        
+        # Outer container
+        width = 680
+        height = 488
+        PAGE.setLineWidth(2)
+        PAGE.setStrokeColorRGB(.97,.41,.12) # Orange
```

Comment:
We discussed this face-to-face, but for posterity, this and related values will need to be "un-hard-coded" and extracted into `cert-data.yml`.